configs/module.go

```diff
@@ -178,16 +178,13 @@ func (m *Module) appendFile(file *File) hcl.Diagnostics {
 	}
 
 	for _, reqd := range file.RequiredProviders {
-		// As an interim *testing* step, we will accept a source argument
-		// but assume that the source is a legacy provider. This allows us to
-		// exercise the provider local names -> fqn logic without changing
-		// terraform's behavior.
+		var fqn addrs.Provider
 		if reqd.Source != "" {
-			// Fixme: once the rest of the provider source logic is implemented,
-			// update this to get the addrs.Provider by using
-			// addrs.ParseProviderSourceString()
+			// FIXME: capture errors
+			fqn, _ = addrs.ParseProviderSourceString(reqd.Source)
+		} else {
+			fqn = addrs.NewLegacyProvider(reqd.Name)
```

Comment:
Is there coverage for this side of the `if`? If not, maybe it's worth adding.

Reply:
If you mean test coverage, it's _absolutely_ missing and coming in the next PR! I'm in an in-between state where I need to remove the `Legacy*` functions from `addrs.Provider`in order to get _real_ tests written, but to do so I need to refactor all of the tests currently using those functions.  

(If you meant something else and I misunderstood, please let me know.)